lib/agents/jsshell.js

```diff
@@ -11,13 +11,19 @@ const stackRe = /^([\s\S]*?)\r?\nStack:\r?\n([\s\S]*)$/;
 const stackFrameRe = /^(.*?)?@(.*?):(\d+):(\d+)?$/;
 
 class JSShell extends ConsoleAgent {
-  async evalScript(code, options = {}) {
-    if (options.module && this.args[0] !== '--module') {
-      this.args.unshift('--module');
+  constructor(options) {
+    super(options);
+    if (this.args.indexOf('--module') !== -1) {
+      throw new Error("Passing --module as a SpiderMonkey host argument is not supported.")
     }
+  }
 
-    if (!options.module && this.args[0] === '--module') {
-      this.args.shift();
+  async evalScript(code, options = {}) {
+    if (options.module) {
+      if (!options.testHostArgs) {
+        options.testHostArgs = [];
+      }
+      options.testHostArgs.push('--module');
     }
 
     return super.evalScript(code, options);
```

Comment:
Rather than mutating options, I think I'd rather have this class override ConsoleAgent `createChildProcess` (which incidentally removes the need to override `evalScript`).
```suggestion
  async createChildProcess(args = [], options = {}) {
    if (options.module) {
      args = ['--module', ...args];
    }
    return super.createChildProcess(args, options);
```